### PR TITLE
Adding JUnit Selenium Sample

### PR DESCRIPTION
**Pull request by TGavan**
creating a pull request for adding sample for Junit and Selenium

### CODE REVIEW

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    void verifyTitleOfHomePage() {
+        //Assertion: Check its title is correct
+        //assertEquals method Parameters: Expected Value, Actual Value, Assertion Message
+        assertTrue(driver.getTitle().contains("TestRail"));
+    }
+
+    @Test
+    void verifyPresenceOfDemoLinkOnHomePage() {
+        //Assertion: Check the presence of demo link
+        WebElement demoLinkElement = driver.findElement(By.cssSelector("a[href='https://us06web.zoom.us/webinar/register/1616784726150/WN_gNYiF3aAQaGewwFfgF-4Ag']"));
```
Comment (bitcoder) [inline samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java:70]: please use a variable for the css selector

Comment (TGavan) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java:70]: @bitcoder unfortunately there is no such attribute which represents the button with label "Get a Demo"

Comment (bitcoder) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java:70]: You can search by the link text:
`WebElement demoLinkElement = driver.findElement(By.linkText("Get a Demo"));`

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        assertTrue(demoLinkElement.isDisplayed());
+    }
+
+    @Test
+    void verifyTheSumOfTwoIntegers() {
```
Comment (bitcoder) [inline samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java:75]: this test should be removed

File: samples/java/junit5-selenium/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# JUnit5 sample project
+
+## Pre-requisites for running the tests in the project
+
+- Java must be installed on your system (JDK8 or higher)
+- Maven must be installed on your system.
+- Python3 must be installed on your system.
+- Chrome browser must be installed on the system for running the tests . Make sure version of chromedriver placed in 'drivers' folder should match the version of locally installed Chrome.
```
Comment (bitcoder) [inline samples/java/junit5-selenium/README.md:8]: maybe here add a reference to where to get the chromedriver.
also tried this out with macos, and it seems that it is looking for the chromedriver under the linux folder

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    @BeforeEach
+    public void setUp() {
+        String os = SystemUtils.OS_NAME;
+        String pathToChromeDriver = null;
+        if (os.contains("Windows")) {
+            pathToChromeDriver = "drivers/windows/chromedriver.exe";
+        } else {
+            pathToChromeDriver = "drivers/linux/chromedriver";
+        }
```
Comment (bitcoder) [inline samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java:31]: maybe using webdrivermanager would make this more clean and less prone to actual details of setting the chromedriver? 

https://github.com/bonigarcia/webdrivermanager

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java (hunk 1 above)
Comment (TGavan) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java]: @bitcoder its done now

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java (hunk 2 above)
Comment (TGavan) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java]: @bitcoder The old test is now removed and its replaced by the new test which I deliberately would like to fail

File: samples/java/junit5-selenium/README.md (hunk 3 above)
Comment (TGavan) [reply samples/java/junit5-selenium/README.md]: @bitcoder as discussed , I added a new folder for mac with appropriate drivers

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java (hunk 4 above)
Comment (TGavan) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java:33]: @bitcoder as discussed , I would  not like to change it . The reason is ,  in CircleCI we are using the image with browser preinstalled on  it and we can also define the particular version of the browser which we would like to install . So in our CircleCI.yaml file , we already fixed chrome version 109 and I have placed appropriate drivers in the drivers folder

**Review by d-rede — CHANGES_REQUESTED** [review]

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java (hunk 4 above)
Comment (TGavan) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java]: @d-rede its removed now

File: samples/java/junit5-selenium/drivers/linux/chromedriver
[the remark this answers is not in the document]
Comment (TGavan) [reply samples/java/junit5-selenium/drivers/linux/chromedriver:1]: @d-rede its removed now

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        options.addArguments("--allow-running-insecure-content");
+        options.addArguments("--ignore-certificate-errors");
+        options.addArguments("--disable-extensions"); // disabling extensions
+        options.addArguments("--disable-dev-shm-usage"); // overcome limited resource problems
+        options.addArguments("--no-sandbox"); // Bypass OS security model
+        options.setExperimentalOption("useAutomationExtension", false);
+        driver = new ChromeDriver(options);
+        driver.manage().window().maximize();
+        driver.manage().timeouts().implicitlyWait(Duration.ofMillis(5000));
+        //Navigation: Open a website
```
[the remark this answers is not in the document]
Comment (TGavan) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java]: @d-rede its done

File: samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    @AfterEach
+    public void tearDown() {
+        //Quit Driver after completing of every test
+        driver.quit();
+    }
+
+    @Test
+    void verifyTitleOfHomePage() {
+        //Assertion: Check its title is correct
+        //assertEquals method Parameters: Expected Value, Actual Value, Assertion Message
```
[the remark this answers is not in the document]
Comment (TGavan) [reply samples/java/junit5-selenium/src/test/java/com/idera/testrail/tests/HomePageTest.java]: @d-rede  this comment in now removed

File: samples/java/junit5-selenium/sample_reports/testrail.jpg
[the remark this answers is not in the document]
Comment (TGavan) [reply samples/java/junit5-selenium/sample_reports/testrail.jpg:1]: @d-rede image is now deleted

**Review by d-rede — APPROVED** [review]

File: samples/java/junit5-selenium/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# JUnit5 sample project
+
+## Pre-requisites for running the tests in the project
+
+- Java must be installed on your system (JDK8 or higher)
```
Comment (d-rede) [inline samples/java/junit5-selenium/README.md:5]: Can we turn this list around? Like:
You need to install the following on your local machine: 
- java
- maven
- etc

File: samples/java/junit5-selenium/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+- Maven must be installed on your system.
+- Python3 must be installed on your system.
+- Chrome browser must be installed on the local system for running the tests.
+
+## How to parse the test results to testrail without using any CI/CD tool
```
Comment (d-rede) [inline samples/java/junit5-selenium/README.md:10]: Don't use "parse results to TestRail", always use import/submit/send in this context.
Also, **TestRail** 😄 
And there's no need to bring CI/CD into the readme. Say from your local machine if you must.

File: samples/java/junit5-selenium/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+- Replace the placeholders in `trcli-config.yml` with your TestRail instance details
+- Execute the commands on the script below if you would like to parse test results without any CI/CD tool
```
Comment (d-rede) [inline samples/java/junit5-selenium/README.md:13]: Same as https://github.com/gurock/automation-frameworks-integration/pull/5/files/1d708df389059031b2f7be7b6d7c7bce4e456ce2#r1158770970

File: samples/java/testng-selenium/README.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# TestNG sample project
```
Comment (d-rede) [inline samples/java/testng-selenium/README.md:1]: See comments in JUnit5 readme

File: samples/java/testng-selenium/sample_reports/TestRail suite/TestRail Tests .html [hunk 11]
```diff
@@ -0,0 +1,145 @@
+<html>
```
Comment (d-rede) [inline samples/java/testng-selenium/sample_reports/TestRail suite/TestRail Tests .html:1]: Can we remove everything under the TestRail suite folder? Or do these have any purpose?

File: samples/java/testng-selenium/testng.xml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (d-rede) [inline samples/java/testng-selenium/testng.xml:1]: Is this file necessary?

File: samples/java/junit5-selenium/.circleci/config.yml [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+          paths:
+            - target/TEST-junit-jupiter.xml
+  install-trcli-and-parse-results:
+    docker:
+      - image: cimg/python:3.11
+    steps:
+      - attach_workspace:
+          at: ./automation-framework-integration
+      - run:
+          name: Install TR CLI and upload the test results to Testrail
```
Comment (d-rede) [inline samples/java/junit5-selenium/.circleci/config.yml:32]: *TestRail

File: samples/java/testng-selenium/testng.xml (hunk 12 above)
Comment (TGavan) [reply samples/java/testng-selenium/testng.xml:1]: @d-rede yes , we need this file to run the tests sequentially

File: samples/java/junit5-selenium/.circleci/config.yml (hunk 13 above)
Comment (TGavan) [reply samples/java/junit5-selenium/.circleci/config.yml]: @d-rede its done now

File: samples/java/junit5-selenium/README.md (hunk 8 above)
Comment (TGavan) [reply samples/java/junit5-selenium/README.md]: @d-rede its done now

File: samples/java/junit5-selenium/README.md (hunk 7 above)
Comment (TGavan) [reply samples/java/junit5-selenium/README.md]: @d-rede its done now

File: samples/java/testng-selenium/README.md (hunk 10 above)
Comment (TGavan) [reply samples/java/testng-selenium/README.md:1]: @d-rede its done now
